filesystem/msystem

```diff
@@ -20,56 +20,10 @@ unset MINGW_CHOST
 unset MINGW_PREFIX
 unset MINGW_PACKAGE_PREFIX
 
-case "${MSYSTEM}" in
-    MINGW32)
-        MSYSTEM_PREFIX='/mingw32'
-        MSYSTEM_CARCH='i686'
-        MSYSTEM_CHOST='i686-w64-mingw32'
-        MINGW_CHOST="${MSYSTEM_CHOST}"
-        MINGW_PREFIX="${MSYSTEM_PREFIX}"
-        MINGW_PACKAGE_PREFIX="mingw-w64-${MSYSTEM_CARCH}"
-        export MSYSTEM_PREFIX MSYSTEM_CARCH MSYSTEM_CHOST MINGW_CHOST MINGW_PREFIX MINGW_PACKAGE_PREFIX
-        ;;
-    MINGW64)
-        MSYSTEM_PREFIX='/mingw64'
-        MSYSTEM_CARCH='x86_64'
-        MSYSTEM_CHOST='x86_64-w64-mingw32'
-        MINGW_CHOST="${MSYSTEM_CHOST}"
-        MINGW_PREFIX="${MSYSTEM_PREFIX}"
-        MINGW_PACKAGE_PREFIX="mingw-w64-${MSYSTEM_CARCH}"
-        export MSYSTEM_PREFIX MSYSTEM_CARCH MSYSTEM_CHOST MINGW_CHOST MINGW_PREFIX MINGW_PACKAGE_PREFIX
-        ;;
-    CLANG64)
-        MSYSTEM_PREFIX='/clang64'
-        MSYSTEM_CARCH='x86_64'
-        MSYSTEM_CHOST='x86_64-w64-mingw32'
-        MINGW_CHOST="${MSYSTEM_CHOST}"
-        MINGW_PREFIX="${MSYSTEM_PREFIX}"
-        MINGW_PACKAGE_PREFIX="mingw-w64-clang-${MSYSTEM_CARCH}"
-        export MSYSTEM_PREFIX MSYSTEM_CARCH MSYSTEM_CHOST MINGW_CHOST MINGW_PREFIX MINGW_PACKAGE_PREFIX
-        ;;
-    CLANGARM64)
-        MSYSTEM_PREFIX='/clangarm64'
-        MSYSTEM_CARCH='aarch64'
-        MSYSTEM_CHOST='aarch64-w64-mingw32'
-        MINGW_CHOST="${MSYSTEM_CHOST}"
-        MINGW_PREFIX="${MSYSTEM_PREFIX}"
-        MINGW_PACKAGE_PREFIX="mingw-w64-clang-${MSYSTEM_CARCH}"
-        export MSYSTEM_PREFIX MSYSTEM_CARCH MSYSTEM_CHOST MINGW_CHOST MINGW_PREFIX MINGW_PACKAGE_PREFIX
-        ;;
-    UCRT64)
-        MSYSTEM_PREFIX='/ucrt64'
-        MSYSTEM_CARCH='x86_64'
-        MSYSTEM_CHOST='x86_64-w64-mingw32'
-        MINGW_CHOST="${MSYSTEM_CHOST}"
-        MINGW_PREFIX="${MSYSTEM_PREFIX}"
-        MINGW_PACKAGE_PREFIX="mingw-w64-ucrt-${MSYSTEM_CARCH}"
-        export MSYSTEM_PREFIX MSYSTEM_CARCH MSYSTEM_CHOST MINGW_CHOST MINGW_PREFIX MINGW_PACKAGE_PREFIX
-        ;;
-    *)
-        MSYSTEM_PREFIX='/usr'
-        MSYSTEM_CARCH="$(/usr/bin/uname -m)"
-        MSYSTEM_CHOST="${MSYSTEM_CARCH}-pc-msys"
-        export MSYSTEM_PREFIX MSYSTEM_CARCH MSYSTEM_CHOST
-        ;;
-esac
+_msystem_d="/etc/msystem.d/$(echo "${MSYSTEM}" | LC_ALL=C /usr/bin/tr A-Z a-z)"
```

Comment:
This creates 3 new processes on every shell startup, no? Can we optimize this?

Reply:
I don't know.  I did it that way because msystem is sourced from profile, and that was the way I found that would work on any POSIX shell.  I just found another way that seems like it would work without subprocesses - `typeset -l`.  But it is also not strictly POSIX, though it should work with bash and ksh so it's probably reasonably portable.

Reply:
hm ok. We already have various shell specific things in profile, so that is always an option too.

What if we just uppercase the filenames?

Reply:
> What if we just uppercase the filenames?

That'd be the easiest, if we're ok with the "ugliness" of the uppercase filenames.  Then `shell` can use a bash-ism to uppercase its parameter, since it already requires bash.  msys2_shell.cmd would already need to uppercase to set MSYSTEM if/when that gets updated.

Reply:
hm, yeah.

I measured and this line takes ~30ms here.

Reply:
uppercased the filenames